src/murfey/server/ispyb.py

```diff
@@ -21,6 +24,39 @@
 )
 
 
+class TransportManager:
+    def __init__(self, transport_type):
+        self.transport = workflows.transport.lookup(transport_type)()
+        self.transport.connect()
+
+    def start_dc(self, message):
+        message["ispyb_command"] = "insert_data_collection"
+        visit = message["visit"]
+        session = (
+            Session()
+            .query(_BLSession, _Proposal)
+            .join(_Proposal, _Proposal.proposalId == _BLSession.proposalId)
+            .options(
+                Load(_BLSession).load_only("sessionId", "visit_number", "proposalId"),
+                Load(_Proposal).load_only(
+                    "proposalId", "proposalCode", "proposalNumber"
+                ),
+            )
+            .filter(
+                sqlalchemy.func.concat(
+                    _Proposal.proposalCode,
+                    _Proposal.proposalNumber,
+                    "-",
+                    _BLSession.visit_number,
+                )
+                == visit
+            )
+        )
+        message["session_id"] = session.first()[0].sessionId
```

Comment:
Either extract this as a "get session for visit" if we need this - or more sensibly - if we need the session id then we should extend the `Visit` model in `util.models` (for that merge the current `main` into your branch first) to hold that extra piece of information and then there's no need to look it up specifically.